Fix dev environment/testing install instructions

sgkasselau opened the pull request:
Hi,

I've put together a few fixes to the development environment/testing installation instructions.

Particularly clarifying how to:

- install Prover9 (needed for the inference doctest);
- get set up in pipenv (useful if you have many projects, and don't want to contaminate your user-level Pip setup);
- get Gitflow set up, including how to start a branch using it;
- how to run the test suite successfully via Tox (particularly in pipenv.)

(I struggled with this on a new machine, so figured I should document it for others!)

Included also is:

- a small fix to a doctest, inference.doctest, to allow it to work with Prover9 2009-11A; and,
- a small enhancement to tox.ini to allow individual tests to be run

Hope this is useful to you!

sgkasselau committed:
Allow tox to run individual tests

sgkasselau committed:
Clarified CONTRIBUTING.md with Prover9 install instructions, pipenv alternative steps, more Gitflow usage/elaborations, and more accurate tox installation steps.

sgkasselau committed:
Make inference.doctest work with Prover9 2009-11A, which seems to have a slightly different output to that expected.

sgkasselau committed:
Add name to AUTHORS.md

tomaarsen commented:
Hey!
To clarify, the prover9-related tests should automatically skip if prover9 is not installed. Most users won't be making changes to this, and thus I wouldn't consider installing prover9 a requirement for new developers. That said, the documentation changes are definitely useful, although I happen to use a different module for my virtual environments.

On that note, I'm not sure whether we should go that detailed with it. Although it helps completely new developers, simply recommending users to use some virtual environment tool might be preferable, as more experienced developers can use whatever tool they are familiar with.

The same goes with git-flow. People have personal preferences here, and perhaps it's best to mention the simplest solution (i.e. as right now, I believe). I'm willing to hear others' thoughts on this though.

sgkasselau committed:
Merge branch 'develop' into feature/fix_test_install_instructions

sgkasselau commented:
Hi Tom!

Thanks for getting back to me!

To explain my choices a bit more, I added the instructions for Prover9 as there didn't seem to be any instructions for this alongside the code itself on how to do this. (And, well, inference is where I'm currently developing other changes, so I wanted this to be clear as possible so that my later work yet-to-come is more accessible.)

Pipenv instructions are elsewhere in CONTRIBUTING.md already, so I added instructions for pipenv where there were existing pip related ones for consistency's sake. So that one environment management tool wasn't being favoured over the other.

I added elaborating documentation on git-flow because CONTRIBUTING.MD already mentioned that the project uses it, and I felt it needed more explanation on how to get started with it. NLTK has a slightly non-standard branching model (i.e. there's no main or master branch) so I wanted to provide the right instructions to make this work alongside the other git-flow documentation.

Hope this helps explain my motivations better?

(Please let me know if you'd like me to withdraw some or all of the CONTRIBUTING.md documentation changes?)

sgkasselau committed:
Merge branch 'develop' into feature/fix_test_install_instructions

alvations commented:
@sgkasselau apologies for the hiatus. Thanks for the updating the contribution, give newer content, this https://github.com/nltk/nltk/pull/3526 should supersede  this PR.
